src/sqlfluff/core/dialects/dialect_bigquery.py

```diff
@@ -224,3 +286,37 @@ class DatatypeSegment(BaseSegment):
             ),
         ),
     )
+
+
+@bigquery_dialect.segment()
+class StructSegment(BaseSegment):
+    """Container of ordered fields each with a type (required) and field name (optional).
+
+    Note that here we're not trying to parse the Struct datatype (struct<...>)
+    which is covered in DatatypeSegment, but the function to construct a struct.
```

Comment:
The word "function" here seems a bit confusing to me. Would it be more appropriate to say "syntax to construct a struct" or "syntax for a struct literal", something kinda like that? And maybe change the segment name as well.

Reply:
sure! I actually took the word directly from the BigQuery documentation https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#constructing_a_struct but I agree it's not very clear, I'll rename

Reply:
I think I made it clearer now! I realized this specific expression is called "Typeless struct" because it allows you to create a struct with inferred types